Testing/SystemTests/tests/framework/SampleWithShapeAndMaterialSaveNexusTest.py

```diff
@@ -0,0 +1,151 @@
+# Mantid Repository : https://github.com/mantidproject/mantid
+#
+# Copyright &copy; 2025 ISIS Rutherford Appleton Laboratory UKRI,
+#   NScD Oak Ridge National Laboratory, European Spallation Source,
+#   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+# SPDX - License - Identifier: GPL - 3.0 +
+import subprocess
+import sys
+import tempfile
+import os
+import json
+from systemtesting import MantidSystemTest
+from mantid.simpleapi import Load
+
+
+class SetupMixin:
+    def setUp(self):
+        temp_dir = tempfile.gettempdir()
+        self.fname = "test_file.nxs"
+        self.save_fpath = os.path.join(temp_dir, self.fname)
+
+        self.temp_dir_literal = json.dumps(temp_dir)
+        self.fname_literal = json.dumps(self.fname)
+        self.save_path_literal = json.dumps(self.save_fpath)
+
+    def execTest(self, test_script):
+        script_path = os.path.join(tempfile.gettempdir(), "subtest_script.py")
+        with open(script_path, "w") as f:
+            f.write(test_script)
+
+        result = subprocess.run(
+            [sys.executable, script_path],
+            capture_output=True,
+            text=True,
+        )
+
+        print("Subprocess STDOUT:\n", result.stdout)
+        print("Subprocess STDERR:\n", result.stderr)
+        self.result_code = result.returncode
+
+    def validate(self):
+        self.assertEqual(self.result_code, 0, "Subprocess crashed or failed assertion")
+        self.assertTrue(os.path.exists(self.save_fpath), "Expected output file not created")
+        try:
+            Load(Filename=self.save_fpath, OutputWorkspace="loaded_ws")
+        except Exception as e:
+            self.fail(f"Failed to load saved file: {e}")
+        return True
+
+
+class TestSetMaterialThenShapeThenSave(SetupMixin, MantidSystemTest):
```

Comment:
I think instead of this class (and the other tests) inheriting from both `SetupMixin` and `MantidSystemTest` could `SetupMixin` inherit from `MantidSystemTest`. I'm actually a bit surprised this works (assuming these tests are running).